KarhooUISDK/CommonUI/Checkbox/CheckboxView.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  CheckboxView.swift
+//  KarhooUISDK
+//
+//  Created by Aleksander Wedrychowski on 04/02/2022.
+//  Copyright © 2022 Flit Technologies Ltd. All rights reserved.
+//
+
+import UIKit
+
+class CheckboxView: UIControl {
+
+    // MARK: - Nested types
+
+    // Some ui properties are not default, so need to describe it properly. This is because visible view is smaller than proper UX expects.
+    enum CustomConstants {
+        static let visibleIconSideLenght: CGFloat = 18
+        static let sideLenght: CGFloat = UIConstants.Dimension.Button.small
+        static let visibleAndActualWidthOffset: CGFloat = sideLenght - visibleIconSideLenght
+    }
+
+    private enum SelectionState {
+        case selected
+        case unselected
+
+        var image: UIImage {
+            switch self {
+            case .selected: return .uisdkImage("checkbox_selected")
+            case .unselected: return .uisdkImage("checkbox_unselected")
+            }
+        }
+    }
+
+    // MARK: - Properties
+
+    override var isSelected: Bool {
+        get {
+            selectionState == .selected
+        }
+        set {
+            updateState(isSelected: newValue)
+        }
+    }
+
+    public var showsError: Bool = false {
+        didSet {
+            updateErrorState()
+        }
+    }
+
+    private var selectionState: SelectionState = .unselected
+
+    // MARK: Views
+
+    private let imageViewContainer = UIView().then {
+        $0.translatesAutoresizingMaskIntoConstraints = false
+        $0.layer.cornerRadius = UIConstants.CornerRadius.xSmall
+        $0.layer.masksToBounds = true
+    }
+
+    private let imageView = UIImageView().then {
+        $0.translatesAutoresizingMaskIntoConstraints = false
+        $0.contentMode = .scaleAspectFit
+    }
+
+    // MARK: - Lifecycle
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setup()
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        translatesAutoresizingMaskIntoConstraints = false
+        isAccessibilityElement = true
+
+        setupHierarchy()
+        setupLayout()
+        setupGesture()
+        updateState(isSelected: false)
+    }
+
+    private func setupHierarchy() {
+        addSubview(imageViewContainer)
+        imageViewContainer.addSubview(imageView)
+    }
+
+    private func setupLayout() {
+        heightAnchor.constraint(equalToConstant: CustomConstants.sideLenght).isActive = true
+        widthAnchor.constraint(equalToConstant: CustomConstants.sideLenght).isActive = true
```

Comment:
You can combine the height and width anchor setting by using` anchor(width: CustomConstants.sideLenght, height: CustomConstants.sideLenght)`